R/session_check_reauth.R

```diff
@@ -0,0 +1,61 @@
+#'@title Checks current session and re-authenticates if necessary
+#'
+#'@inheritParams authenticate_sb
+#'@param session (Advanced) Alternatively supply a specific session to be checked
+#'@param ... Any additional parameters are ignored. 
+#'
+#'@description
+#'This checks the state of your Sciencebase session and, if invalid or if you've supplied
+#'a new user name, it will re-authenticate. This is primarily a helper function for 
+#'packages that reverse-depend on sbtools.
+#'
+#'@examples
+#'
+#'\dontrun{
+#'
+#'#will prompt for username/pass if needed
+#'session_check_reauth()
+#'
+#'session_check_reauth('user@@usgs.gov')
+#'
+#'#default behavior
+#'session_check_reauth(session=current_session())
+#'
+#'}
+#'
+#'
+#'@export
+session_check_reauth = function(username, password, session, ...){
```

Comment:
so use this in place of authenticate_sb, right? call it just before running any sbtools commands that require authentication? or do sbtools functions that need authentication already call this themselves?